src/components/NcAvatar/NcAvatar.vue

```diff
@@ -121,8 +121,10 @@ export default {
 		:tabindex="hasMenu ? '0' : undefined"
 		:aria-label="avatarAriaLabel"
 		:role="hasMenu ? 'button' : undefined"
-		@click="toggleMenu"
-		@keydown.enter="toggleMenu">
+		v-on="hasMenu ? {
+			click: toggleMenu,
+			keydown: toggleMenu,
+		} : null">
```

Comment:
IMO, we can move `hasMenu` check to the events handler `toggleMenu`. Then we wouldn't need to implement Vue's functionality manually and template would be clearer without 4-lines computing directive.

Reply:
No I explicitly moved it to the template, otherwise the listeners would always be added.
But when there is no menu we should not have any listeners, because that element is not interactive.